### PR TITLE
Fix `pod install --project-directory=ios` failing when Hermes is enabled

### PR DESCRIPTION
**Pull request by tido64**
## Summary

`pod install --project-directory=ios` fails when Hermes is enabled:

```
% pod install --project-directory=ios
[Codegen] Generating ios/build/generated/ios/React-Codegen.podspec.json
[Hermes] Extracting Hermes tarball (5244f8)

[DEBUG] pwd: /~/example
[DEBUG] cp: cp ../node_modules/react-native/sdks/hermes-engine.podspec ../node_modules/react-native/sdks/hermes/hermes-engine.podspec

warn Multiple Podfiles were found: ios/Podfile,macos/Podfile. Choosing ios/Podfile automatically. If you would like to select a different one, you can configure it via "project.ios.sourceDir". You can learn more about it here: https://github.com/react-native-community/cli/blob/master/docs/configuration.md
Auto-linking React Native module for target `ReactTestApp`: ReactTestApp-DevSupport
Analyzing dependencies
Fetching podspec for `DoubleConversion` from `../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec`
[Codegen] Found FBReactNativeSpec
Fetching podspec for `RCT-Folly` from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`
Fetching podspec for `boost` from `../node_modules/react-native/third-party-podspecs/boost.podspec`
Fetching podspec for `glog` from `../node_modules/react-native/third-party-podspecs/glog.podspec`
[!] No podspec found for `hermes-engine` in `../node_modules/react-native/sdks/hermes/hermes-engine.podspec`
```

## Changelog

[iOS] [Fixed] - `pod install --project-directory=ios` fails when Hermes is enabled

## Test Plan

Running `pod install --project-directory=ios` is enough to trigger this issue.

### CODE REVIEW

File: scripts/react_native_pods.rb [hunk 1]
```diff
@@ -621,20 +624,20 @@ def downloadAndConfigureHermesSource(react_native_path)
...
   end
   Pod::UI.puts "[Hermes] Extracting Hermes tarball (#{hermes_tag_sha.slice(0,6)})"
-  system("tar -zxf #{hermes_tarball_path} --strip-components=1 --directory #{hermes_dir}")
+  system("tar -zxf #{hermes_tarball_path.to_s} --strip-components=1 --directory #{hermes_dir.to_s}")
 
   # Use React Native's own scripts to build Hermes
-  system("cp #{sdks_dir}/hermes-engine/hermes-engine.podspec #{hermes_dir}/hermes-engine.podspec")
-  system("cp #{sdks_dir}/hermes-engine/utils/* #{hermes_dir}/utils/.")
+  FileUtils.cp(sdks_dir.join("hermes-engine", "hermes-engine.podspec"), hermes_dir.join("hermes-engine.podspec"))
+  FileUtils.cp_r(sdks_dir.join("hermes-engine", "utils"), hermes_dir)
```
Comment (tido64) [inline scripts/react_native_pods.rb:638]: This section changed since the 0.69 cut. The PR for 0.69 is here: #33819

**Comment by tido64** [comment]
> For context, what's the failure?

Sorry, forgot to add the error message here. I posted it over here: https://github.com/reactwg/react-native-releases/discussions/21#discussioncomment-2738239. In any case, I've updated the PR description.

**Comment by tido64** [comment]
@cortinico: It looks like this code was removed in favour of a JS script. I'm closing this PR, but the one for `0.69-stable` is still required.

File: scripts/react_native_pods.rb (hunk 1 above)
Comment (cortinico) [reply scripts/react_native_pods.rb:638]: Thanks I've tracked this 👍

**Pull request closed without merging**